src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+    private final ColorSupplier color;
```

Comment:
The 'ColorSupplier' should not be stored as a field in the 'Square' class. Instead, pass the 'Color' object directly to the constructor.